core/src/main/java/jenkins/ClassLoaderReflectionToolkit.java

```diff
@@ -34,7 +34,7 @@
             gCLL = ClassLoader.class.getDeclaredMethod("getClassLoadingLock", String.class);
             gCLL.setAccessible(true);
         } catch (NoSuchMethodException x) {
-            // OK, Java 6
+            throw new AssertionError(x);
```

Comment:
Not supposed to happen with Java 7+ JDK, so it now throws an `AssertionError` like I saw some lines above  instead of ignoring it.